src/frontend/actions/user-actions/user-actions.ts

```diff
@@ -26,8 +26,13 @@ export class UserActions {
   ) {}
 
   /// Toggle the expansion state of a node
-  toggle(node: Node) {
-    switch (this.viewState.expandState(node)) {
+  toggle(node: Node, defaultState: ExpandState) {
+    let state = this.viewState.expandState(node);
+    if (state == null) {
+      state = defaultState;
```

Comment:
No, because the default expansion state changes depending on what level/depth the node is at. It must be passed from `NodeItem`.